source/core/slang-digest-util.h

```diff
@@ -0,0 +1,35 @@
+// slang-digest-utils.h - Utility functions specifically designed to be used with slang::Digest
+#pragma once
+#include "../../slang.h"
+#include "slang-string.h"
+
+namespace Slang
+{
+    using slang::Digest;
+
+    struct DigestUtil
+    {
+        // Compute the digest for an UnownedStringSlice
+        static Digest computeDigestForStringSlice(UnownedStringSlice text);
+
+        // Combines the two provided digests.
+        static Digest combine(const Digest& digestA, const Digest& digestB);
+
+        // Returns the hash stored in digest as a String.
+        static String toString(const Digest& digest);
+
+        // Returns the hash represented by hashString as a Digest.
+        static Digest fromString(UnownedStringSlice hashString);
+    };
+
+    inline StringBuilder& operator<<(StringBuilder& sb, const Digest& d)
+    {
+        for (Index i = 0; i < 4; ++i)
+        {
+            char temp[8];
+            IntToAscii(temp, d.values[i], 16);
```

Comment:
You will need to call `sb.append(uint8Values[i[)`, otherwise it is going to produce results in a different endian order.

Reply:
Probably need a test for this in the similar way as `toString` and `fromString`, or you can simply change the implementation of `toString` to call this.

Reply:
Yeah, the implementation here should just be exactly what is existing in `toString`, and then make `toString` call this.

Reply:
I mean, if you make toString call `operator <<` then you don't need a dedicated test for it.